### DISCUSSION TITLE
How do I delegate focus and keyboard navigation to the first items of a nested list in grid lists ?

### DISCUSSION THREAD

**Discussion by maximeoger** [adobe/react-spectrum#6797] [Q&A]
Hello,

I use the useGridList hook to handle nested list object that renders a sub list of items in a recursive way.

When I navigate with the mouse, everything works as expected, but when I use the keyboard navigation, an error occurs : 

`Cannot read properties of null (reading 'focus')`

I think this is because the focus does not persist on the sub list items. Is there some kind of way to interrupt the keyboard navigation in the main list and move it to the sub list ?

[Here is a working example](https://codesandbox.io/p/sandbox/react-aria-components-nested-gridlist-focus-6flry3?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clz8i9afs00063b6mgcpo6d5x%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clz8i9afr00023b6m52mvrqe6%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clz8i9afr00033b6mmzf5zbji%2522%257D%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clz8i9afr00053b6mvcdt7uov%2522%257D%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clz8i9afr00023b6m52mvrqe6%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clz8i9afr00013b6metcnqgok%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.js%2522%257D%255D%252C%2522id%2522%253A%2522clz8i9afr00023b6m52mvrqe6%2522%252C%2522activeTabId%2522%253A%2522clz8i9afr00013b6metcnqgok%2522%257D%252C%2522clz8i9afr00053b6mvcdt7uov%2522%253A%257B%2522id%2522%253A%2522clz8i9afr00053b6mvcdt7uov%2522%252C%2522activeTabId%2522%253A%2522clz8k70h400jk3b6mbwv5zqcx%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clz8i9afr00043b6mmg39yccx%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522clz8jjclh00an3b6mn4rpwrti%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522clz8jmyii00f23b6mzopu97i6%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522clz8joqb000he3b6m8vbplgd7%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522clz8k70h400jk3b6mbwv5zqcx%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522clz8i9afr00033b6mmzf5zbji%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clz8i9afr00033b6mmzf5zbji%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by snowystinger** [comment]
From a quick peek at what you're doing, it looks similar to submenus. Have you had a look through our Submenu implementation in React Spectrum? https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/menu/src/SubmenuTrigger.tsx that may shed some light on what's going on for you.
